datasync/chaindatafetcher/kafka/config.go

```diff
@@ -74,14 +80,22 @@ func GetDefaultKafkaConfig() *KafkaConfig {
 		Replicas:             DefaultReplicas,
 		SegmentSizeBytes:     DefaultSegmentSizeBytes,
 		MaxMessageNumber:     DefaultMaxMessageNumber,
+		MsgVersion:           DefaultKafkaMessageVersion,
+		ProducerId:           GetDefaultProducerId(),
 	}
 }
 
+func GetDefaultProducerId() string {
+	rb := common.MakeRandomBytes(8)
+	randomString := hexutil.Encode(rb)
+	return DefaultProducerIdPrefix + randomString[2:]
```

Comment:
Can you explain why you use randomString[2:]? I'm just curious.

Reply:
@yoomee1313 In order to give an unique producer id, I put random hex string without prefix `0x`.

Reply:
understood, thanks!